azure_functions_worker/bindings/meta.py

```diff
@@ -94,5 +108,35 @@ def to_outgoing_proto(binding: str, obj: typing.Any, *,
             f'unable to encode outgoing TypedData: '
             f'unsupported type "{binding}" for '
             f'Python type "{type(obj).__name__}"')
+    return datum
+
 
+def to_outgoing_proto(binding: str, obj: typing.Any, *,
+                      pytype: typing.Optional[type]) -> protos.TypedData:
+    datum = get_datum(binding, obj, pytype)
     return datumdef.datum_as_proto(datum)
+
+
+def to_outgoing_param_binding(binding: str, obj: typing.Any, *,
+                              pytype: typing.Optional[type],
+                              out_name: str,
+                              shmem_mgr) \
+        -> protos.ParameterBinding:
+    datum = get_datum(binding, obj, pytype)
+    shared_mem_value = None
+    # If shared memory is enabled, try to transfer to host over shared memory
```

Comment:
Not sure I fully understand (so not changing anything yet.)
The first if will check if shared memory is enabled and supported - if so, it will "try" to use it. This is the default case.
If either the pervious attempt to use shared memory fail, or it was disabled/unsupported, we use RPC. 

Maybe you can propose the change and if it makes sense I can accept it from here?